.kokoro/downstream-client-library-check.sh

```diff
@@ -18,6 +18,9 @@ set -eo pipefail
 set -x
 
 CLIENT_LIBRARY=$1
+
+# Example JOB_TYPE: "test" or "graalvm"
+export JOB_TYPE=$2
```

Comment:
This variable is not used?

Reply:
Yes, as you noticed, the JOB_TYPE env variable is not getting picked up. The initial line `./kokoro/build.sh` was able to run the tests in either standard Java or with native image compilation based on the env variable (as seen in the jobs run in the [first commit](https://github.com/googleapis/java-shared-dependencies/pull/771/commits/cad9e434a2d2e8d14e9a14f8b5660280de5e328a) )but modifying it to be in original state still forces it to run with standard java. It looks like this same change needs to be made in gax as the most recent runs of the downstream checks have been skipping the native image compilation stage. For example: https://github.com/googleapis/gax-java/runs/7619427005?check_suite_focus=true

Reply:
Ah sorry my previous statement was kinda confusing. We need two types of downstream workflows (represented by downstream-native-image and downstream.yaml). While the call to `downstream-client-library-check.sh ` in downstream.yaml sets the JOB_TYPE env variable to "test", the call to `downstream-client-library-check.sh` in downstream-native-image.yaml will set the JOB_TYPE to "graalvm". `downstream-client-library-check.sh` will then use those values to do `export JOB_TYPE=${arg_value}`. We will need to change the script to call `build.sh` which runs the respective commands based on the `JOB_TYPE` present.